### DISCUSSION TITLE
[Question] How do I use this with IRedMail

### DISCUSSION THREAD

**Discussion by Ryu945** [nginx-proxy/nginx-proxy#2560] [Q&A]
From what I understand is going on, this communicates to containers over HTTP.  I am using:  https://github.com/lejmr/iredmail-docker

When NGINX tries to communicate over HTTP, IRedMail redirects to HTTPs but NGINX can only communicate to containers over HTTP so it makes an endless loop.  I tried:

VIRTUAL_PROTO=https
VIRTUAL_PORT=443

in the IRedMail container.  This not only did not work and gave a wrong certificate error.  It broke the other working containers like the NextCloud container this ships with even though the only setting that was changed was in the IRedMail container.  You can see which Nextcloud container here:

https://blog.ssdnodes.com/blog/installing-nextcloud-docker/

I am not sure why this is happening and how to fix it.
 
edit:
I figured out most of the problems and I got  to the point of a login page.  What I don't know is what ports I need open.

**Comment by cron410** [comment]
You would want to make the NGINX container your HTTPS termination point, not iredmail if possible.
